cloud.foundry.cli/src/main/java/cloud/foundry/cli/services/DumpController.java

```diff
@@ -0,0 +1,86 @@
+package cloud.foundry.cli.services;
+
+import static picocli.CommandLine.usage;
+import static picocli.CommandLine.Command;
+import static picocli.CommandLine.Mixin;
+
+import cloud.foundry.cli.crosscutting.mapping.RefResolver;
+import cloud.foundry.cli.crosscutting.mapping.beans.ConfigBean;
+import cloud.foundry.cli.crosscutting.mapping.beans.SpecBean;
+
+import java.util.concurrent.Callable;
+
+/**
+ * This class realizes the functionality that is needed for the dump commands.
+ * It defines command classes that take care of the command execution. They are derived from the {@link DumpCommandBase}
+ * class that contains common behaviour of all the subcommands.
+ */
+@Command(name = "dump", header = "%n@|green Resolve " + RefResolver.REF_KEY + "-occurrences in a configuration file" +
+        " and print the result.|@",
+        mixinStandardHelpOptions = true, subcommands = {
+        DumpController.DumpAllCommand.class,
+        DumpController.DumpServicesCommand.class,
+        DumpController.DumpApplicationsCommand.class,
+        DumpController.DumpSpaceDevelopersCommand.class
+})
+public class DumpController implements Callable<Integer> {
+
+    // FIXME this field is actually not necessary, but needed due to the argument extension mechanism in the class
+    // CfArgumentsCreator... its contents are simply ignored by the dump command
```

Comment:
Have you been able to come up with some way to resolve this FIXME? I am pretty sure Java reflections could help.

Reply:
No, I have no idea for any workaround. The CfArgumentsCreator always extends the entered arguments to contain api-host, organization and space (which are not needed for this command). Unless the LoginCommandOptions are defined as Mixin, calling any dump command always results in an error due to the unnecessary arguments.

Reply:
Thanks for your reply. If you can, please leave the PR open, I'll try to find a few minutes to check it out myself.